### ISSUE TITLE
Server-Side rendering support?

### ISSUE THREAD

**Issue by quoininc-huudatran** [wojtekmaj/react-pdf#678]
**Before you start - checklist**

 - [X] I have read documentation in README
 - [X] I have checked [sample](https://github.com/wojtekmaj/react-pdf/tree/master/sample) and [test](https://github.com/wojtekmaj/react-pdf/tree/master/test) suites to see real life basic implementation
 - [X] I have checked if this question is not already asked

**What are you trying to achieve? Please describe.**

I would like to do server-side-rendering. It will not render on the server-side the pdf part, but we still get the error:

```
(node:23431) UnhandledPromiseRejectionWarning: ReferenceError: window is not defined
node_modules/react-pdf/node_modules/pdfjs-dist/lib/web/ui_utils.js:564:3
```

**Describe solutions you've tried**

There are some dependency that are expecting to be a browser (like `window`). I've tried to define `global.window`, but I think the context of the service worker doesn't define it.

**Additional information**

If applicable, add screenshots (preferably with browser console open) and files you have an issue with to help explain your problem.

**Environment**

 - Browser (if applicable) [e.g. Chrome 57, Firefox 59]:  Node v12.18.3
 - React-PDF version [e.g. 3.0.4]:  react-pdf@5.0.0
 - React version [e.g. 16.3.0]: react@16.14.0
 - Webpack version (if applicable) [e.g. 4.16.2]: webpack@4.44.2 (but not used for SSR)

**Comment by wojtekmaj** [comment]
There's absolutely no way this would work server side. I'm sorry!

**Comment by quoininc-huudatran** [comment]
I am not LOOKING to have it render server side, i want to be able to compile my page and when it gets on the browser, it renders it. There are already peerDependencies not declared.

My code is to click and a modals open to display the PDF. So it will NEVER be rendered on server side, but it needs to be able to compile. I don't understand why, even when i define `global.window`, i still get the error mentioned.

**Comment by wojtekmaj** [comment]
Unfortunately these global.window declarations are in pdf.js and not React-PDF. Given that this will never start working on server side, i doubt Mozilla would be interested in a fix.

I suggest loading a mock on server side conditionally.

**Comment by quoininc-huudatran** [comment]
Apparently, this is related to #657 which I didn't check because I don't use Gatsby. I will try a 4.x and see if it works.

**Comment by gyto23** [comment]
@quoininc-huudatran just asking if your error still persist? I have identical issue as you have but I am using the nextJS but I am not rendering the page on server side which is for me is odd because my pdf is only called when I am calling for.

On the other hand, @wojtekmaj have you try to run pdf document on the React.Suspence with React.lazy?

**Comment by quoininc-huudatran** [comment]
@gyto23 I ended up just trying v4 since v5 is still new and I'll let it go some miles (see that other issue i linked above).

But I was also unable to use `import ... from...` syntax on the top, but I bypass it by hiding it deep in the code and use `require()` so that it doesn't pop up until that code is running (on the browser, which is the context where it is expected to run). I got no tests on that specific component, so that might come up later.
